File: Java - my completed version/src/main/java/com/gildedrose/ItemUpdater.java
package com.gildedrose;

public final class ItemUpdater {

    private ItemUpdater() {
    }

    public static void update(Item item) {
        if (item instanceof AgedBrie) {
            ((AgedBrie) item).updateQuality();
        } else if (item instanceof Backstage) {
            ((Backstage) item).updateQuality();
        } else if (item instanceof Sulfuras) {
            ((Sulfuras) item).updateQuality();
        } else if (item instanceof Conjured) {
            ((Conjured) item).updateQuality();
        } else if (item instanceof Normal) {
            ((Normal) item).updateQuality();
        }
    }
}

File: TypeScript - my completed version/app/item-updater.ts
import type { Item } from '#app/gilded-rose';

export const AGED_BRIE = 'Aged Brie';
export const BACKSTAGE_PASSES = 'Backstage passes to a TAFKAL80ETC concert';
export const CONJURED = 'Conjured';
export const SULFURAS = 'Sulfuras, Hand of Ragnaros';

export interface ItemUpdater {
  update(item: Item): void;
}
